core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -628,6 +628,24 @@ private[spark] class Executor(
           }
       } finally {
         runningTasks.remove(taskId)
+        if(conf.getBoolean("spark.conf.GCOnIdle.enabled", false)) {
```

Comment:
The config should be more like `spark.memory.idleGCThreshold` or something.
Also, check it once and save the value; no need to check every time.